org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DefaultVariableProvider.java

```diff
@@ -76,20 +78,24 @@ private void initialize() {
 
         registerValueFormatter(new StringObjectFormatter(), 2);
         registerValueFormatter(new ArrayObjectFormatter(this::typeToString), 2);
-        registerValueFormatter(new ClassObjectFormatter(this::typeToString), 2);
+        registerValueFormatter(new ClassObjectFormatter(this::typeToString), 2);        
+        
     }
-
-    private static IFormatter getFormatter(Map<? extends IFormatter, Integer> formatterMap, Type type,
-                                           Map<String, Object> options) {
-        List<? extends IFormatter> formatterList =
-                formatterMap.keySet().stream().filter(t -> t.acceptType(type, options))
-                        .sorted((a, b) ->
-                                -Integer.compare(formatterMap.get(a), formatterMap.get(b))).collect(Collectors.toList());
-        if (formatterList.isEmpty()) {
-            throw new UnsupportedOperationException(String.format("There is no related formatter for type %s.",
-                    type == null ? "null" : type.name()));
+    
+    @Override
+    public Map<String, Object> getDefaultOptions() {
+        if (defaultOptions == null) {
+            defaultOptions = new HashMap<>();
+            valueFormatterMap.keySet().forEach(this::mergeDefaultOptions);
+            typeFormatterMap.keySet().forEach(this::mergeDefaultOptions);
         }
-        return formatterList.get(0);
+        return defaultOptions;
+    }
+    
+
+    @Override
+    public IValueFormatter getValueFormatter(Type type, Map<String, Object> options) {
```

Comment:
On setValue, when we have a type and a String value, we need the related formatter to restore the JDI value for calling setValue in JDI.